File: python/数组/252. 会议室.py
from typing import List
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort(key=lambda x:(x[0],x[1]))
        i = 1
        while i < len(intervals):
            if intervals[i][0] < intervals[i-1][1]:
                return False
            i+=1
        return True
s = Solution()
interval = [[0,30],[5,10],[15,20]]
print(s.canAttendMeetings(interval))

File: go/array/252. 会议室.go
package array

import (
	"sort"
)

func canAttendMeetings(intervals [][]int) bool {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	i := 1
	for i < len(intervals) {
		if intervals[i][0] < intervals[i-1][1] {
			return false
		}
		i++
	}
	return true
}
